packages/ledger/src/converters/ledger.converters.ts

```diff
@@ -1,19 +1,57 @@
 import { toNullable } from "@dfinity/utils";
-import type { TransferArg } from "../../candid/icrc1_ledger";
-import type { TransferParams } from "../types/ledger.params";
+import type {
+  ApproveArgs,
+  TransferArg,
+  TransferFromArgs,
+} from "../../candid/icrc1_ledger";
+import type {
+  ApproveParams,
+  TransferFromParams,
+  TransferParams,
+} from "../types/ledger.params";
 
 export const toTransferArg = ({
-  to,
   from_subaccount,
   fee,
-  amount,
   created_at_time,
   memo,
+  ...rest
 }: TransferParams): TransferArg => ({
-  to,
-  amount,
+  ...rest,
   fee: toNullable(fee),
   memo: toNullable(memo),
   from_subaccount: toNullable(from_subaccount),
   created_at_time: toNullable(created_at_time),
 });
+
+export const toTransferFromArgs = ({
```

Comment:
Do we want to convert back and forth all this types?

Shouldn't we stop doing that and just pass the same type to the clients?

We started doing that in sns-js

Reply:
We already do - i.e. it's not new, I just followed the existing pattern of `toTransferArgs`.

Reply:
Yes, I know. Because we got it from nns-js.

I'm just wondering which way we want to go. We have some inconsistency between nns-js and sns-js at the moment. Which way is the correct one?

Reply:
As discussed offline. We go on with the PR and the question will be treated or discussed separatly.